exercises/little-schemer/canonical-data.json

```diff
@@ -0,0 +1,215 @@
+        },
+        {
+          "description": "null of non-list",
+          "property": "exception",
+          "input": {
+            "function": "null?",
+            "list": "'Zang'"
+          },
+          "comments": [
+            "The primitive null? is defined only for lists."
+          ],
+          "expected": -1
+        }
+      ]
+    },
+    {
+      "description": "Test car of a list",
+      "cases": [
+        {
+          "description": "car of simple list",
+          "property": "equality",
+          "input": {
+            "function": "car",
+            "list": [1]
+          },
+          "expected": 1
+        },
+        {
+          "description": "car of small list",
+          "property": "equality",
+          "input": {
+            "function": "car",
+            "list": ["Immaculate", 23, "G.O.A.T"]
+          },
+          "expected": "'Immaculate'"
+        },
+        {
+          "description": "car of nested list",
+          "property": "equality",
+          "input": {
+            "function": "car",
+            "list": [[4, 5], 6, 7, [8, 9]]
+          },
+          "expected": [4, 5]
+        },
+        {
+          "description": "car of null list",
+          "property": "exception",
+          "input": {
+            "function": "car",
+            "list": []
+          },
+          "comments": [
+            "The primitive car is defined only for non-empty lists."
+          ],
+          "expected": -1
+        }
+      ]
+    },
+    {
+      "description": "Test cdr of a list",
+      "cases": [
+        {
+          "description": "cdr of simple list",
+          "property": "equality",
+          "input": {
+            "function": "cdr",
+            "list": [1]
+          },
+          "expected": []
+        },
+        {
+          "description": "cdr of small list",
+          "property": "equality",
+          "input": {
+            "function": "cdr",
+            "list": ["Immaculate", 23, "G.O.A.T"]
+          },
+          "expected": [23, "G.O.A.T"]
+        },
+        {
+          "description": "cdr of nested list",
+          "property": "equality",
+          "input": {
+            "function": "cdr",
+            "list": [[4, 5], 6, 7, [8, 9]]
+          },
+          "expected": [6, 7, [8, 9]]
+        },
+        {
+          "description": "cdr of null list",
+          "property": "exception",
+          "input": {
+            "function": "cdr",
+            "list": []
+          },
+          "comments": [
+            "The primitive cdr is defined only for non-empty lists. The cdr of any non-empty list is always another list"
+          ],
+          "expected": -1
+        }
+      ]
+    },
+    {
+      "description": "Test atom of a list",
+      "cases": [
+        {
+          "description": "atom of a list",
+          "property": "boolean",
+          "input": {
+            "function": "atom?",
+            "list": ["Immaculate", 23]
```

Comment:
since it is expected that inputs to `atom?` may be things other than lists, I would say to use something reflective of that fact. Perhaps `input`, or if that's deemed too redundant (having an `input` nested inside `input`) then `value`.